clients/docs/c.rst

```diff
@@ -0,0 +1,156 @@
+-----
+
+#. Clone the `confluentinc/examples GitHub repository
+   <https://github.com/confluentinc/examples>`__ and check out the
+   :litwithvars:`|release|-post` branch.
+
+   .. codewithvars:: bash
+
+      git clone https://github.com/confluentinc/examples
+      cd examples
+      git checkout |release|-post
+
+#. Change directory to the example for C.
+
+   .. code-block:: bash
+
+      cd clients/cloud/c/
+
+#. .. include:: includes/client-example-create-file.rst
+
+
+Basic Producer and Consumer
+----------------------------
+
+In this example, the producer writes JSON data to a topic in your Kafka cluster.
+Each record has a key representing a username (for example, ``alice``) and a
+value of a count, in JSON format (for example, ``{"count": 0}``). The consumer
+reads the same topic and keeps a rolling sum of the counts as it processes each
+record.
+
+
+Produce Records
+~~~~~~~~~~~~~~~
+
+#. Run the producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka
+     cluster
+   - the topic name
+
+   .. code-block:: bash
+
+      ./producer test1 $HOME/.confluent/librdkafka.config
+
+#. Verify the producer sent all the messages. You should see:
+
+   .. code-block:: bash
+
+      Creating topic test1
+      Topic test1 successfully created
+      Producing message #0 to test1: alice={ "count": 1 }
+      Producing message #1 to test1: alice={ "count": 2 }
+      Producing message #2 to test1: alice={ "count": 3 }
+      Producing message #3 to test1: alice={ "count": 4 }
+      Producing message #4 to test1: alice={ "count": 5 }
+      Producing message #5 to test1: alice={ "count": 6 }
+      Producing message #6 to test1: alice={ "count": 7 }
+      Producing message #7 to test1: alice={ "count": 8 }
+      Producing message #8 to test1: alice={ "count": 9 }
+      Producing message #9 to test1: alice={ "count": 10 }
+      Waiting for 10 more delivery results
+      Message delivered to test1 [0] at offset 0 in 22.75ms: { "count": 1 }
+      Message delivered to test1 [0] at offset 1 in 22.77ms: { "count": 2 }
+      Message delivered to test1 [0] at offset 2 in 22.77ms: { "count": 3 }
+      Message delivered to test1 [0] at offset 3 in 22.78ms: { "count": 4 }
+      Message delivered to test1 [0] at offset 4 in 22.78ms: { "count": 5 }
+      Message delivered to test1 [0] at offset 5 in 22.78ms: { "count": 6 }
+      Message delivered to test1 [0] at offset 6 in 22.78ms: { "count": 7 }
+      Message delivered to test1 [0] at offset 7 in 22.79ms: { "count": 8 }
+      Message delivered to test1 [0] at offset 8 in 22.80ms: { "count": 9 }
+      Message delivered to test1 [0] at offset 9 in 22.81ms: { "count": 10 }
+      10/10 messages delivered
+
+#. View the :devx-examples:`producer code|clients/cloud/c/producer.c`.
+
+Consume Records
+~~~~~~~~~~~~~~~
+
+#. Run the consumer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka
```

Comment:
For this example, we need to switch the order of these two bullet points (first `the topic name`, second `the local file...`) to match the code

Reply:
updated